rosserial_arduino/src/ros_lib/examples/TcpBlink/TcpBlink.ino

```diff
@@ -0,0 +1,47 @@
+/*
+ * rosserial Subscriber Example using TCP on Arduino Shield (Wiznet W5100 based)
+ * Blinks an LED on callback
+ */
+#include <SPI.h>
+#include <Ethernet.h>
+
+#define ROSSERIAL_ARDUINO_TCP
```

Comment:
I'm not over the moon about switching behaviour based on this define, though it doesn't kind of follow with what we've done previously, particularly around Leonardo and selecting whether to use the UART or USB connection.

I suppose the only alternatives are supplying a `ros.h` with custom NodeHandle typedefs in each example/firmware, which doesn't really make sense (though it's what we do internally at Clearpath).

Reply:
I agree it is not the best thing. I thing it should be better to include a different header file directly in the sketch. But it means revise the architecture of rosserial_client.

I don't see how you manage the typedef from the firmware, do you have a simple example?

Reply:
In our product firmwares, we're using rosserial ports that are not public— for example, that talk to the rosserial_server UDP implementation. In upstream rosserial (here), the `ros.h` file is supplied with the generated ros_lib and therefore has to be generic, but our `ros.h` files are much more use-case specific since they're instead supplied with the firmware source tree.

Anyway, it's a bit academic. I think what you have is fine for now, but it would be great to figure out how to document better for future users some suggested design patterns.

Reply:
About documentation, I think I will edit the rosserial_arduino wiki page to list all the available switches (USBCON, HW_SERIAL...).

About changing the design pattern, it would be nice to give a more explicit API but I have no idea how to do it right now.